2-mandatory/3-shopping-list.js

```diff
@@ -19,9 +19,24 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+  let newObject = {};
+  newObject.name = recipe.name;
+  let missingItems = [];
+  for (element of recipe.ingredients) {
+    if (!pantry.fridgeContents.includes(element) && (!pantry.cupboardContents.includes(element))) {
+      missingItems.push(element);
+    }
+  }
```

Comment:
great job, you can also concat two arrays inside of the pantry and check if the items in the recipe are in that array or not.
this my own solution but feel free to solve it the way you are more comfortable, just wanted to give you some other ideas
`function createShoppingList(recipeObj){
  let result = {name: recipeObj.name , shoppingList: []}
  let allIHave = pantry.fridgeContents.concat(pantry.cupboardContents)
  recipeObj.ingredients.map(eachIng => {if(!allIHave.includes(eachIng)){
    result.shoppingList.push(eachIng)
  }})
   return result;
}`